design-documents/stateless-api-token.md

```diff
@@ -0,0 +1,178 @@
+## Introduce stateless web API tokens
```

Comment:
`Stateless tokens` is a confusing term by itself. In fact, the proposed token will contain a state, so it makes the naming even more confusing. My understanding is that this proposal is more about client-side sessions using JWT.

The main benefit of the solution is the possibility to avoid DB queries and save some DB capacity. However, the proposed solution doesn't eliminate them completely. Authorization part (user roles, resources, permissions) should still be located on the server side, so we are trading one DB query to CPU power spent on encryption/decryption. It's not clear if this benefit actually exists for the proposed scenario.

Regarding long-living tokens(API tokens), I clearly see benefit if we use this technology for customer sessions, because they contain more information, they are short-living(there is no need to revoke tokens, so we can get more benefit) 


The API in this proposal is highly coupled with another proposal about JWT. Probably it makes sense to merge them.


In addition to token revocation, we need to describe the rotation of secrets or asymmetric keys. This is Adobe security requirements

Reply:
* removed the `Stateless tokens` term
 
* there are 2 main benefits: not having to store and clean up tokens / do not compromise security by tokens being readable from DB
 
* added proposed JWT library and tried to merge API proposed in the JWT proposal with the one here
 
* asymmetric algos will be usable with JWT TokenEncoder implementations but have no practical use with JWS used for authentication - that's why I don't mention them. Rotation of secrets would be out of scope of this proposal since we use secrets for other stuff and the problem should be addressed separately

Reply:
👍 Looks good to me. One more occurrence of "stateless tokens" in file name

Reply:
fixed